store/postgres/src/relational_queries.rs

```diff
@@ -532,7 +561,9 @@ impl<'a> QueryFilter<'a> {
     ) -> QueryResult<()> {
         let column = self.column(attribute);
 
-        if column.is_text() && value.is_string() {
+        if column.is_fulltext() {
+            PrefixComparison::new(Comparison::Match, column, value).walk_ast(out.reborrow())?;
```

Comment:
I don't think we want to go through the `PrefixComparison` machinery here which is just a workaround for the fact that we allow arbitrarily large strings in `text` columns, but only index a prefix of them. It should be possible to turn fulltext searches directly into something like `<column> @@ <value>` like we do for other types.